src/test/java/com/microsoft/jenkins/acs/integration/DCOSDeploymentIT.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.jenkins.acs.integration;
+
+import com.microsoft.azure.management.compute.ContainerServiceOchestratorTypes;
+import com.microsoft.jenkins.acs.ACSDeploymentBuilder;
+import com.microsoft.jenkins.acs.ACSDeploymentContext;
+import hudson.model.Result;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.docker.commons.credentials.DockerRegistryEndpoint;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static com.microsoft.jenkins.acs.integration.TestHelpers.loadFile;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+public class DCOSDeploymentIT extends IntegrationTest {
+    @ClassRule
+    public static ACSRule acs = new ACSRule(ContainerServiceOchestratorTypes.DCOS.toString());
+
+    private ACSDeploymentContext deploymentContext;
+
+    @Before
+    @Override
+    public void setup() throws Exception {
+        super.setup();
+        setupJenkinsCredentials(acs);
+
+        deploymentContext = new ACSDeploymentContext(
+                azureCredentialsId,
+                acs.resourceGroup,
+                acs.resourceName + " | DCOS",
+                sshCredentialsId,
+                "dcos/*.json");
+    }
+
+    @Test
+    public void testSingleDirectDeployment() throws Exception {
+        loadFile(getClass(), workspace, "dcos/direct.json");
+
+        new ACSDeploymentBuilder(deploymentContext).perform(run, workspace, launcher, taskListener);
+
+        verify(run, never()).setResult(Result.FAILURE);
+    }
+
+    @Test
+    public void testPrivateDockerRegistry() throws Exception {
+        if (StringUtils.isBlank(testEnv.dockerUsername)) {
```

Comment:
Should fail if dockerUsername is empty?

Reply:
This check was added to exclude the private repository test so that we don't have to create ACR instance. 

As a complete integration test we would better fail on empty username. I will update it.